server/data/data-cassandra/src/main/java/org/apache/james/dlp/eventsourcing/cassandra/DLPConfigurationItemAddedDTO.java

```diff
@@ -0,0 +1,112 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.dlp.eventsourcing.cassandra;
+
+import static org.apache.james.dlp.eventsourcing.cassandra.DLPConfigurationItemDTO.fromDTOs;
+
+import java.util.List;
+import java.util.Objects;
+
+import org.apache.james.dlp.eventsourcing.aggregates.DLPAggregateId;
+import org.apache.james.dlp.eventsourcing.events.ConfigurationItemsAdded;
+import org.apache.james.eventsourcing.EventId;
+import org.apache.james.eventsourcing.eventstore.cassandra.dto.EventDTO;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
+
+class DLPConfigurationItemAddedDTO implements EventDTO {
+
+    @JsonIgnore
+    public static DLPConfigurationItemAddedDTO from(ConfigurationItemsAdded event, String type) {
+        return new DLPConfigurationItemAddedDTO(
+            type,
+            event.eventId().serialize(),
+            event.getAggregateId().asAggregateKey(),
+            DLPConfigurationItemDTO.from(event.getRules()));
+    }
+
+    private final String type;
+    private final int eventId;
+    private final String aggregateId;
+    private final List<DLPConfigurationItemDTO> configurationItems;
+
+    @JsonCreator
+    private DLPConfigurationItemAddedDTO(
+            @JsonProperty("type") String type,
+            @JsonProperty("eventId") int eventId,
+            @JsonProperty("aggregateId") String aggregateId,
+            @JsonProperty("configurationItems") List<DLPConfigurationItemDTO> configurationItems) {
```

Comment:
I don't think you need these JsonProperty annotations when the field has the same name than the property

Reply:
When have to use `@JsonCreator` to guide Jackson, It requires `@JsonProperty` specified in each argument except you did registers an parameter name support module.

[Why when a constructor is annotated with @JsonCreator, its arguments must be annotated with @JsonProperty?
](https://stackoverflow.com/questions/21920367/why-when-a-constructor-is-annotated-with-jsoncreator-its-arguments-must-be-ann/33578478#33578478)

Reply:
ok, strange because we have some places in the code where we use it with JsonProperty